app/serializers/concept_serializer.rb

```diff
@@ -0,0 +1,24 @@
+class ConceptSerializer < ActiveModel::Serializer
+  attributes :id, :name, :uid, :parent_id, :level, :parent_uid
+  delegate :cache_key, :to => :object
+
+  def to_json(*args)
+    Rails.cache.fetch expand_cache_key(self.class.to_s.underscore, cache_key, 'to-json') do
```

Comment:
@marcellosachs The SQL execution time isn't the problem, it's actually rendering the whole thing to JSON that seems to take awhile. At least that's what I get from the logs. This is not the real example, but it's something like this, where the `Views` time is much, much higher than `ActiveRecord`.

```
Completed 200 OK in 788ms (Views: 741.6ms | ActiveRecord: 4.2ms)
```

You can try it for yourself, try firing up the postgres console on staging and running the query that gets generated by Concept.all_with_level. It's very fast.